### DISCUSSION TITLE
Userscripts for some ongoing issues

### DISCUSSION THREAD

**Discussion by okiehsch** [uBlockOrigin/uAssets#21766] [General]
We endorse/support some userscripts for ongoing hard to fix issues, for example https://github.com/List-KR/microShield.
We do this in various issues, it would be better to add all of them in a note in our readme or a pinned issue imo.
What do you all think?

**Comment by stephenhawk8054** [comment]
I agree. Having a common place for users to see which to add/solve which sites would be good.

**Comment by ghajini** [comment]
one of the ongoing issue , antiadblock on `dropgalaxy.com` & `devuploads.com`
a Very Usefull Userscript is developed by a developer @seia-soto ,thanks for your help & contribution
Please find userscript at `https://seia-soto.github.io/userscripts/`

- Install a userscript manager such as violentmonkey or tampermonkey
- Click on **Devuploads Defuser** at `https://seia-soto.github.io/userscripts/`
- This will install corresponding userscript

**Reply by garry-ut99** [reply]
Another one of ongoing issue - circumvention ads on `chip.de` can be also fixed by installing https://github.com/AdguardTeam/AdGuardExtra userscript : https://github.com/uBlockOrigin/uAssets/issues/21504#issuecomment-1879054564

**Comment by stephenhawk8054** [comment]
@gorhill What do you think about this? Currently we are recommending [pixeltris' userscripts](https://github.com/pixeltris/TwitchAdSolutions#scripts) for Twitch on various support threads (github, reddit...) anyway. Would tickboxes/toggles in the settings for these scripts be better for users?

**Reply by ItsProfessional** [reply]
> Would tickboxes/toggles in the settings for these scripts be better for users?

Counter argument: That's introducing remote code execution into uBO, unless gorhill is willing review to each revision of the script(s) before they go live for uBO users (which is unlikely for understandable reasons).

**Reply by stephenhawk8054** [reply]
I mean it's the same as when users use `userResourcesLocation` for years now, especially as we are already instructing them how to add the userscripts for Twitch case.

We are not enabling it by default, there's no RCE without users' knowing and consents of using it.

**Reply by ItsProfessional** [reply]
If it's added, there should be a confirmation with warnings about RCE and that the author of the extension isn't responsible for any form of damage when users enable those scripts.

**Reply by garry-ut99** [reply]
It was already declined in the past: a Wiki [commit](https://github.com/uBlockOrigin/uBlock-issues/wiki/Resources-Library/_compare/48b31b44465b6fccaaf14ee6475fd8034d522853...42857e316fb9e28d70988edb80b5812b3f61bdd0) with  links to community scriptlets was reverted by another [commit](https://github.com/uBlockOrigin/uBlock-issues/wiki/Resources-Library/_compare/42857e316fb9e28d70988edb80b5812b3f61bdd0...be672e3cff0592e9052d3bc4e81be5dfd8b2b7a9) (even despite one of authors of the scriptlets was a uBO team member):

![ss](https://github.com/uBlockOrigin/uAssets/assets/72945564/2878e226-d43e-44fa-a25f-fddd50a771de)

Given: "_The documentation is not the place to maintain a list of external source of scriptlets_. " - then following the same logic: uBO is even more not such place either, unless like already mentioned, each script revision would be reviewed manually by uBO creator, which would require time and effort and can be considered "burden of maintentain" or unless changed mind and the logic does not apply anymore nowadays in 2024 given more and more websites require 3rd-party scripts solution, also it's worth to remember what happened with Nano in the past: 

<details>
<summary>Spoiler</summary>

- https://github.com/jspenguin2017/Snippets/issues/5
- https://github.com/NanoAdblocker/NanoCore/issues/362
- https://github.com/jspenguin2017/Snippets/issues/2

</details>

**Reply by ghost** [reply]
https://github.com/uBlockOrigin/uAssets/discussions/21766#discussioncomment-8914786 is more unambiguous and noticeable for a future use-case where u don't have to run this in a offline setup to check for Twitch userscripts. Or expect someone to complain about it then more then more then more, now is the 2nd best time to prevent that, says some ancient chinese proverb about trees.

Thanks

**Comment by ghost** [comment]
I strongly recommend not using userscripts. And personally I need to change some passwords since I installed the first extension that was recommended above violentmonkey or something it simply doesn't work for me even though I tested it and then added some not incorrect `@match` lines.

Thanks for posting, now I know what to never do with browser extensions that are not regularly vetted.

PS - All this for Twitch where rarely anything of value is provided except copius amounts of porn. Or toxic noise.

**Reply by MasterKia** [reply]
> I need to change some passwords since I installed the first extension that was recommended

Any evidence to back up your claim?
